feat: add initial sponsorship docs (#190)
reown-com/reown-docs: docs/appkit/recipes/switching-to-send-calls.mdx

## Switching to Send Calls

With Smart Account wallets on the rise, new standards are being adopted by Dapps
and Wallets that take advantage of their capabilities. One of these standards

> being [EIP 5792](https://eips.ethereum.org/EIPS/eip-5792). In short, it

Reviewer: ```suggestion
is [EIP 5792](https://eips.ethereum.org/EIPS/eip-5792). In short, it
```

> introduces a few methods that smart accounts would support that enable a couple

Reviewer: ```suggestion
introduces a few methods that smart accounts would support, that enable a couple
```

of cool features including:
1. **Batched Transactions**: This allows to batch multiple transactions into a
single one that will be executed. 
2. **Using Paymasters**: EIP 5792 has a concept of `capabilities` which determine
what a wallet can do and also what a dapp wants a wallet to do. This allows the
use of Paymasters, which are defined in [ERC 7677](https://eips.ethereum.org/EIPS/eip-7677). 

### A note on Smart Accounts and EOAs

> Not all users will be using Smart Accounts and will be using EOAs (Externally

Reviewer: ```suggestion
Not all users will be using Smart Accounts; Some will be using EOAs (Externally
```

> Owned Accounts) and will be using EOAs (Externally Owned Accounts). These

Reviewer: ```suggestion
Owned Accounts). These
```

wallets do not support EIP 5792 and will be incompatible with `sendCalls`.

> This means that before using `sendCalls`, there needs to be a check on whether

Reviewer: ```suggestion
This means that before using `sendCalls`, there a need to check on whether
```

Author: This is not grammatically correct

> or not it is compatible. Thankfully, this can be achieved using

Reviewer: ```suggestion
the account is compatible or not. Thankfully, this can be achieved using
```

`getCapabilities`. 

An example custom hook that leverages Wagmi's hooks for `getCapabilities` can be
found

> [here](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/hooks/useWagmiActiveCapabilities.ts#L22).

Reviewer: ```suggestion
[in our lab](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/hooks/useWagmiActiveCapabilities.ts#L22).
```

This hook is then used

> [here](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx#L40).

Reviewer: ```suggestion
also [in our lab](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx#L40).
```

> Essentially - the Wallet will inform the Dapp if it supports EIP 5792.

Reviewer: ```suggestion
Essentially - the Wallet will inform the Dapp if it supports EIP 5792 or not.
```

allow sponsoring gas.

### Sending transactions

For Dapps that are using `writeContractAsync`, switching to `sendCalls`
involves a minor change in encoding the call. An example of it in use can be
found
[here](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiSendCallsWithPaymasterServiceTest.tsx#L137).
Important to note is that the function call data is pre-encoded
[on line 12](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiSendCallsWithPaymasterServiceTest.tsx#L12)
within the same file. This is the main difference as `writeContractAsync`
encodes the calls itself. 

### Getting Receipts or Call Status

Typically, alongside `writeContractAsync` or similar calls, there would be a
`waitForTransactionReceipt` used to wait for the transaction. Sending calls via
the EIP 5792 standars by design does not return a transaction hash by design. 

Instead, `sendCalls` will return an identifier for the transaction which can be
queried via getting the calls' status. In Wagmi's case, this is done using
[`useCallsStatus`](https://wagmi.sh/react/api/hooks/useCallsStatus#usecallsstatus).

> An example of using that can be found

Reviewer: ```suggestion
An example can be found
```

> [here](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx#L52).

Reviewer: ```suggestion
[in our lab](https://github.com/reown-com/appkit/blob/main/apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx#L52).
```
